Write up full build instructions

INSTALL is generic, and starts from the ./configure phase. 

I had success eventually with a combination of `autoreconf`, `automake --install-missing` and `libtoolize`, albeit with warnings. 

Please update INSTALL with full build instructions.

The issue was closed but the problem is unsolved, as the instructions are still sorely lacking.

Tested using b2ba61e
I looked at https://github.com/linuxmint/pix/blob/master/INSTALL#L37

    The file `configure.ac' (or `configure.in') is used to create
    `configure' by a program called `autoconf'.  You only need
    `configure.ac' if you want to change it or regenerate `configure' using
    a newer version of `autoconf'.

No mention at all of what to do with `autoconf`.

```bash
git clone https://github.com/linuxmint/pix.git code-pix
cd code-pix
autoconf
```
results in a whole bunch of errors:
```text
configure.ac:13: error: possibly undefined macro: AM_INIT_AUTOMAKE
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.ac:26: error: possibly undefined macro: AM_MAINTAINER_MODE
configure.ac:42: error: possibly undefined macro: AM_PROG_CC_C_O
configure.ac:101: error: possibly undefined macro: AC_DEFINE
configure.ac:125: error: possibly undefined macro: AM_CONDITIONAL
configure.ac:256: error: possibly undefined macro: AM_GLIB_GNU_GETTEXT
```
and then proceeding to the next step results in more errors
```bash
$ ./configure
./configure: line 2164: syntax error near unexpected token `1.11'
./configure: line 2164: `AM_INIT_AUTOMAKE(1.11 foreign no-dist-gzip dist-xz tar-ustar subdir-objects)'
$ make
make: *** No targets specified and no makefile found.  Stop.
```

Not to mention that the instructions lead to an in-source build (well, the instructions currently lead to no build at all), while best practices have for years recommended out-of-source builds.

--- a/INSTALL
+++ b/INSTALL
@@ -42,7 +42,9 @@ a newer version of `autoconf'.
 The simplest way to compile this package is:
 
   1. `cd' to the directory containing the package's source code and type
-     `./configure' to configure the package for your system.  If you're
+     `./autogen.sh` to generate the autotools build files.
+
+  2. Type `./configure' to configure the package for your system.  If you're
      using `csh' on an old version of System V, you might need to type
      `sh ./configure' instead to prevent `csh' from trying to execute
      `configure' itself.
@@ -50,15 +52,15 @@ The simplest way to compile this package is:
      Running `configure' takes awhile.  While running, it prints some
      messages telling which features it is checking for.
 
-  2. Type `make' to compile the package.
+  3. Type `make' to compile the package.
 
-  3. Optionally, type `make check' to run any self-tests that come with
+  4. Optionally, type `make check' to run any self-tests that come with
      the package.
 
-  4. Type `make install' to install the programs and any data files and
+  5. Type `make install' to install the programs and any data files and
      documentation.
 
-  5. You can remove the program binaries and object files from the
+  6. You can remove the program binaries and object files from the
      source code directory by typing `make clean'.  To also remove the
      files that `configure' created (so you can compile the package for
      a different kind of computer), type `make distclean'.  There is

--- a/autogen.sh
+++ b/autogen.sh
@@ -1,0 +1,41 @@
+#!/bin/sh
+# Run this to generate all the initial makefiles, etc.
+test -n "$srcdir" || srcdir=$(dirname "$0")
+test -n "$srcdir" || srcdir=.
+
+olddir=$(pwd)
+
+cd $srcdir
+
+(test -f configure.ac) || {
+    echo "*** ERROR: Directory '$srcdir' does not look like the top-level project directory ***"
+    exit 1
+}
+
+# shellcheck disable=SC2016
+PKG_NAME=$(autoconf --trace 'AC_INIT:$1' configure.ac)
+
+if [ "$#" = 0 -a "x$NOCONFIGURE" = "x" ]; then
+    echo "*** WARNING: I am going to run 'configure' with no arguments." >&2
+    echo "*** If you wish to pass any to it, please specify them on the" >&2
+    echo "*** '$0' command line." >&2
+    echo "" >&2
+fi
+
+mkdir -p m4
+
+glib-gettextize --force --copy || exit 1
+intltoolize --force --copy --automake || exit 1
+autoreconf --verbose --force --install || exit 1
+
+cd "$olddir"
+if [ "$NOCONFIGURE" = "" ]; then
+    $srcdir/configure "$@" || exit 1
+
+    if [ "$1" = "--help" ]; then exit 0 else
+        echo "Now type 'make' to compile $PKG_NAME" || exit 1
+    fi
+else
+    echo "Skipping configure process."
+fi
+

